api/python/ai/chronon/cli/tui/widgets.py

```
@@ -0,0 +1,573 @@
+            return
+            
+        # Check if step is visible in current viewport
+        viewport_start = self.h_scroll_offset
+        viewport_end = self.h_scroll_offset + self.viewport_width
+        
+        # If step is not visible, scroll to show it
+        if step_start_idx < viewport_start:
+            self.h_scroll_offset = max(0, step_start_idx - 5)  # Add some padding
+        elif step_end_idx >= viewport_end:
+            self.h_scroll_offset = min(step_end_idx - self.viewport_width + 5, 
+                                     len(self.date_list) - self.viewport_width)
```

Flake8 (7.2.0)

[error] continuation line under-indented for visual indent

(E128)